Src/Newtonsoft.Json/Utilities/JavaScriptUtils.cs

```diff
@@ -314,5 +313,175 @@ public static string ToEscapedJavaScriptString(string value, char delimiter, boo
                 return w.ToString();
             }
         }
+
+#if !(NET20 || NET35 || NET40 || PORTABLE40)
+
+        public static async Task<char[]> WriteEscapedJavaScriptStringAsync(TextWriter writer, string s, char delimiter, bool appendDelimiters,
+    bool[] charEscapeFlags, StringEscapeHandling stringEscapeHandling, IArrayPool<char> bufferPool, char[] writeBuffer, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // leading delimiter
+            if (appendDelimiters)
+            {
+                await writer.WriteAsync(delimiter, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
In the other branch there's a check hit within the call to `writer.WriteAsync`, and it seemed a good idea to have one near the beginning of the call in the case that that branch wasn't taken.

Co-operative cancellation is one of those things it's impossible to be perfect with. I think this one is reasonable, but there are a few others in this method that seem excessive, so I'll cull them.